lib/core/services/upload.service.ts

```diff
@@ -95,6 +95,13 @@ export class UploadService {
         return !!this.activeTask;
     }
 
+    isQueueFinishedUploading(): boolean {
+        const finishedFileStates = [FileUploadStatus.Complete, FileUploadStatus.Cancelled, FileUploadStatus.Aborted, FileUploadStatus.Error, FileUploadStatus.Deleted];
+        return this.queue.reduce((finishedUploading: boolean, currentFile: FileModel) => {
+            return finishedUploading && finishedFileStates.indexOf(currentFile.status) > -1;
+        }, true);
+    }
+
```

Comment:
there is already a method called isUploading() and I think there is no need to create a new one

Reply:
I'd agree, but using that method intermittently causes the button to be reenabled in between one upload having finished and the next one starting. This method takes into consideration the Pending and Starting states too. Basically it's a method to check if the queue still has things left to do or if it's idle, I wouldn't know how better to name it though.

Reply:
thnx

Reply:
just thinking if we really need the first case and why.
probably the isUploading is hiding bugs
can u try to use u code and run the unit test?